canonical-web-and-design/maas-ui
# [Periodic hardware sync] - hardware sync failed notification

## Done

- add a notification for when hardware sync has failed
  - migrate MachineNotifications.test.tsx to testing-library 
  - migrate SummaryNotifications.test.tsx to testing-library

## Screenshots
<img width="1451" alt="image" src="https://user-images.githubusercontent.com/7452681/162965567-ea5a6a83-958b-49ea-b9cd-74d42936588b.png">

`MAAS/r/machine/xxn6s7/summary`

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- go to any non-deployed machine details page and verify that the failed hardware sync notification is not displayed
- go to any deployed machine details page with hardware sync enabled and verify the notification is displayed (e.g. `MAAS/r/machine/xxn6s7/summary` on bolla)

## Fixes

Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/786
Fixes: https://github.com/canonical-web-and-design/maas-ui/issues/3799
Fixes: https://github.com/canonical-web-and-design/maas-ui/issues/3800

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
